### ISSUE TITLE
Human-readable // machine-parsable outputs

### ISSUE THREAD

**Issue by moviuro** [btrfs8-revamp/btrfs8-revamp#6]
Let's face it, humans love columns and tables, machines like `key:value`.

At the time of writing, btrfs is mainly (only?) human-readable.

```
btrfs subvolume create test
btrfs fi usage test
btrfs sub delete test
```

Those outputs are completely machine-fiendly.

I propose the introduction of a switch that forces machine-readable output.
- `--machine`
- `--output=machine|human`
- `-m`

Machine-readable output is characterized by fixed and predictable behavior, easy `grep(1)`-ing and `awk(1)`-ing. We may use here som unique identifiers for numbers and whatnot.

For example:

```
# btrfs -m subvolume create test
PATH    /mnt/btrfs/test
UUID    8c7a6613-e0e9-4b16-a8ac-4cb6f1fc795d
DATE    1445412480
...
```

```
# btrfs -m fi usage test
DEV_SIZE    1820314568
DEV_ALOC    1178564230
DEV_UALOC   ...
...
```

Human readable output is characterized by readability. Using `column(1)` in this spirit might help. Units can be used to avoid ridiculously big/small numbers (1e12 byte vs. 1Tb):

```
$ btrfs fi usage /path # taken from man
Overall:
    Device size:                   1.82TiB
    Device allocated:              1.17TiB
    Device unallocated:          669.99GiB
    Device missing:                  0.00B
    Used:                          1.14TiB
    Free (estimated):            692.57GiB      (min: 692.57GiB)
    Data ratio:                       1.00
    Metadata ratio:                   1.00
    Global reserve:              512.00MiB      (used: 0.00B)
```

EDIT: note the position of the `-m` switch that is set just after `btrfs`: it is intended so as to have an easy alias to `btrfs -m` in scripts (`btrfs() { command btrfs -m "$@" }`)

**Comment by kdave** [comment]
Agreed. This was also asked in the mailinglist. I have summed up my ideas here, https://github.com/kdave/drafts/blob/master/btrfs/progs-output-format.txt basically matches what you propose:
* global option to 'btrfs' with the requested format of output, we can do eg. json or key=value etc

**Comment by Ferroin** [comment]
Ideally, I'd love to se support for both JSON and key=value pairs.  JSON is simpler to handle from high level languages, but key=value pairs are much better from shell scripts (you can just execute the command and import everything as variables).

**Comment by moviuro** [comment]
For JSON output, see https://github.com/jpmens/jo ?

**Comment by kdave** [comment]
Json support partially implemented in 5.2, more output backends possible.
